### PR TITLE
Separate out remote access API from Redux

### PR DESCRIPTION
**Pull request by axelboc**
I've noticed a bunch of issues with the remote access feature. I've fixed a couple of obvious ones (cf. comments) and will open issues for the others.

### CODE REVIEW

File: ui/src/actions/remoteAccess.js [hunk 1]
```diff
@@ -52,154 +42,64 @@ export function requestControl(
...
-      method: 'POST',
-      credentials: 'include',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      body: JSON.stringify({ allow }),
+    sendRespondToControlRequest(giveControl, message).then(() => {
+      dispatch(getLoginInfo());
+      dispatch(getRaState());
```
Comment (axelboc) [inline ui/src/actions/remoteAccess.js:80]: Now dispatching these actions after responding to a control request, so the modal with accept/deny closes itself and the state gets updated right away.

File: ui/src/components/RemoteAccess/ObserverDialog.jsx [hunk 2]
```diff
@@ -32,13 +29,15 @@ export class ObserverDialog extends React.Component {
     );
   }
 
-  accept() {
+  accept(evt) {
+    evt.preventDefault();
```
Comment (axelboc) [inline ui/src/components/RemoteAccess/ObserverDialog.jsx:33]: Entering a name and submitting the form was triggering a full page navigation/reload.

**Comment by marcus-oscarsson** [comment]
Watch out, there seems to be some conflicts !

**Comment by axelboc** [comment]
No worries, just need to `git rebase --onto`. Trying to fix the specs; I'll mark the PR as ready once I've figured it out.

**Comment by axelboc** [comment]
No issues with the specs in the end; the CI had failed to install MXCuBE for some reason but it's passing now. I'm struggling to make the E2E pass locally in Firefox — I'll have to keep investigating.

File: ui/src/actions/remoteAccess.js (hunk 1 above)
Comment (marcus-oscarsson) [reply ui/src/actions/remoteAccess.js:80]: :+1:

Comment (marcus-oscarsson) [reply ui/src/actions/remoteAccess.js:80]: Thanks !

**Comment by marcus-oscarsson** [comment]
Well done, thanks !

**Pull request merged**